### ISSUE TITLE
Player Loses Survival Inventory if They Quit During a Bedwars Game

### ISSUE THREAD

**Issue by panterathehacker** [ScreamingSandals/BedWars#460]
#### Describe the bug
- If a player quits the server while in a bedwars game, and then rejoins the server while the game is still in session, they respawn in the arena but are no longer part of the game (they have their regular survival inventory; they are not counted as a player in the game; they have no team prefix, etc). Then, when that bedwars game does end, and all players all teleported back to the lobby, their entire inventory is wiped, meaning they lose everything in their survival inventory.

- Same thing happens if they leave while spectating a game. When they rejoin, they lose all their survival inventory when the game ends.

#### Server and plugin version
Spigot version: Paper 1.19.3
BedWars version: BedWars v 0.2.27.1

#### Installed plugins on server (can be obtained by `/pl`)
Plugins: https://imgur.com/a/o2FhAh5

#### Console log
No console errors

#### Configuration (of config.yml and shop.yml)

config.yml: https://pastebin.com/ZzNmfDqg
shop.yml: https://pastebin.com/1czWtred

**Comment by Misat11** [comment]
I see you have Multiverse-Inventories installed. This plugin is not very well supported and causes issues when used together with BedWars. I recommend you creating a lobby which shares the inventory with the bedwars world that will be used for joining/leaving the arena and then disable joining using the command from any other location.

**Comment by panterathehacker** [comment]
> I see you have Multiverse-Inventories installed. This plugin is not very well supported and causes issues when used together with BedWars. I recommend you creating a lobby which shares the inventory with the bedwars world that will be used for joining/leaving the arena and then disable joining using the command from any other location.

Sorry not quite sure I follow what you’re suggesting. You’re saying to give the bedwars arena it’s own inventory, shared with the world that the lobby is located in?

**Comment by Misat11** [comment]
Yes, not exactly the lobby of the game itself, but some mainlobby world from which people would join your bedwars games.

**Comment by lukasblacha** [comment]
Same problem here...
Multiverse is installed, but no multivese inventories.
No matter from which location players joins a game, via sign in main lobby or command, their inventory gets cleared...
BedWars Version BedWars 0.2.31
Paper-Server Version 1.20.2

Does some one has a clue?

Installed Prlugins:
AdvancedPortals, AdvancedRegionMarket, ArmShopBridge, AuctionHouse, BedWars, ChestShop, Chunky, Citizens, CMILib, CoreProtect, DeadChest, DecentHolograms, DiscordSRV, dynmap, Dynmap-GriefPrevention, *Dynmap-SimpleClans, Essentials, EssentialsChat, EssentialsSpawn, FastAsyncWorldEdit, floodgate, Geyser-Spigot, GriefPrevention, Harbor, Images, Jobs, LuckPerms, Multiverse-Core, Multiverse-Inventories, OpenInv, PaintballBattle, PlaceholderAPI, PluginPortal, SimpleClans, SurvivalGames, TAB, Vault, ViaVersion, voicechat, VoidGen, Vulcan, WorldEditSUI, WorldGuard
